plugin_system/plugins/base_plugin.py

```diff
@@ -0,0 +1,62 @@
+import pysam
+
+
+class BasePlugin:
+    """
+    Interface for caf generation plugins
+    """
+
+    __is_plugin__ = True
+
+    def create_sample_phenotype_index(self) -> dict[str, list[str] | set[str]]:
+        """given any arbitrary phenotypical data input, return a dict mapping from each sample id to the sample's list of phenotypes
+
+        Returns:
+            dict[str, list[str] | set[str]]: index of a sample id to sample's phenotypes
+        """
+        raise NotImplementedError(
+            "Plugins must implement sample-to-phenotype index method"
+        )
+
+    def include_sample(self, sample_id: str, record: pysam.VariantRecord, phenotype: str, sample_phenotype_index: dict[str, list[str]]) -> bool:
+        """given a sample id, determine whether to include it in the allele counts
+
+        Args:
+            sample_id (str): sample_id used to uniquly identify a sample ID
+            record (pysam.VariantRecord): PySam record object representing a VCF row
+            phenotype (str): phenotype of interest, matching phenotype codes used in sample_phenotype_)index
```

Comment:
I made a note to this effect in `evidence.py` but I wonder if we could simply use this method to define the phenotype of interest rather than receiving it from the calling context.

Reply:
see #105